qiskit/algorithms/gibbs_state_preparation/gibbs_state_builder.py

```diff
@@ -0,0 +1,43 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Interface for building Gibbs States."""
+from abc import abstractmethod
+from typing import Union, Dict, Optional
+
+from qiskit.algorithms.gibbs_state_preparation.gibbs_state_sampler import GibbsStateSampler
+from qiskit.circuit import Parameter
+from qiskit.opflow import OperatorBase
+
+
+class GibbsStateBuilder:
```

Comment:
Do we really need this class? 
Can we structure the code such that the 'GibbsStateSampler' is sufficient?